src/wallet/wallet.cpp

```diff
@@ -1139,6 +1163,82 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlockIndex *pindex,
     }
 }
 
+void CWallet::UpdatedBlockHeaderTip(bool fInitialDownload, const CBlockIndex *pindexNew)
+{
+    int bestSPVBlockHeight = 0;
+    {
+        LOCK2(cs_main, cs_wallet);
+        bestSPVBlockHeight = nBestSpvHeight;
+    }
+
+    // check if the new tip extends the wallets current known best headers tip
+    const CBlockIndex *pCurrent = pindexNew;
+    bool fReorg = true;
+    while (pCurrent)
+    {
+        if (pCurrent->GetBlockHash() == bestSpvBlockHash || bestSpvBlockHash.IsNull())
+        {
+            // new header tip extends the current headers chain, standard operation
+            fReorg = false;
+            break;
+        }
+        pCurrent = pCurrent->pprev;
+    }
+    if (fReorg)
+    {
+        // we are on a different chain, we don't know at which dept the fork happend
+        // do a SPV rescan back to the oldest wtx time that was received from a block that now has been forked away
+        CWalletDB walletdb(strWalletFile, "r+", false);
+        int64_t rescanBackTo = std::numeric_limits<int64_t>::max();
+        // the current wallets best header tip is not in the active headers chain
+        for (map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
+        {
+            CWalletTx* wtx = &(*it).second;
+            if (wtx->fSPV)
+            {
+                bool found = false;
+                const CBlockIndex *pCurrent = pindexNew;
+                while (pCurrent)
+                {
+                    if (wtx->hashBlock == pCurrent->GetBlockHash())
+                    {
+                        found = true;
+                        break;
+                    }
+                    pCurrent = pCurrent->pprev;
+                }
+                if (!found)
+                {
+                    if (wtx->GetTxTime() < rescanBackTo)
+                        rescanBackTo = wtx->GetTxTime();
+                    // wtx is no longer confirmed
+                    wtx->hashBlock.SetNull();
+                    wtx->MarkDirty();
+                    walletdb.WriteTx(*wtx);
+                }
+            }
+        }
+        // try to find a reasonable height to spv-scan from again
+        const CBlockIndex *pCurrent = pindexNew;
+        nBestSpvHeight = 0;
+        bestSpvBlockHash.SetNull();
+        while (pCurrent)
+        {
+            if (pCurrent->GetBlockTime() < rescanBackTo)
+            {
+                nBestSpvHeight = pCurrent->nHeight;
+                bestSpvBlockHash = pCurrent->GetBlockHash();
+                break;
+            }
+            pCurrent = pCurrent->pprev;
+        }
+        if (nBestSpvHeight > 0 && rescanBackTo != std::numeric_limits<int64_t>::max())
+            ScanSPV(rescanBackTo);
+    }
+    if (fUseSPV && pindexNew && pindexNew->nHeight > bestSPVBlockHeight)
```

Comment:
Is this actually supposed to be bestSPVBlockHeight not nBestSpvHeight? Why compare against previous height instead of current? If this is intended would suggest adding a comment to explain and also renaming bestSPVBlockHeight to previousBestSPVBlockHeight to make clear this is supposed to be the previous height.